### DISCUSSION TITLE
Missing guards features

### DISCUSSION THREAD

**Discussion by Archetipo95** [nuxt/nuxt#18224] [Questions] [answered]
How to trigger on client side routing some guards? Middleware for clients is no longer available.

**Comment by wobsoriano** [comment, marked as answer]
I think with the absence of middleware guards, you can add your auth logic in your layouts.

Say you have `layouts/index.vue`

```html
<script setup lang="ts">
    const authStore = useAuthStore()
    const router = useRouter()

    router.beforeEach((to, from) => {
        if (to === '/original/path' && !authStore.isAuthenticated) {
            return '/redirect/to/path'
        }
    })
</script>

<template>
    <div>
        <slot />
    </div>
</template>
```

or if you want to do it by page, you can create an auth checker composable and add it per page.
